Tasks/NuGetCommandV2/nugetrestore.ts

```diff
@@ -163,19 +170,30 @@ export async function run(nuGetPath: string): Promise<void> {
             }
         }
 
-        // Setting creds in the temp NuGet.config if needed
-        await nuGetConfigHelper.setAuthForSourcesInTempNuGetConfigAsync();
+        tl.debug('nuGetConfigPath'); // satus debugging
+        tl.debug(nuGetConfigPath);
 
-        // Use config file if:
-        //     - User selected "Select feeds" option
-        //     - User selected "NuGet.config" option and the nuGetConfig input has a value
-        let useConfigFile: boolean = selectOrConfig === "select" || (selectOrConfig === "config" && !!nuGetConfigPath);
-        let configFile = useConfigFile ? nuGetConfigHelper.tempNugetConfigPath : undefined;
-
-        if (useConfigFile)
-        {
-            credCleanup = () => tl.rmRF(nuGetConfigHelper.tempNugetConfigPath);
+        if (!useV2CredProvider && !configFile) {
+            // Setting creds in the temp NuGet.config if needed
+            await nuGetConfigHelper.setAuthForSourcesInTempNuGetConfigAsync();
+            tl.debug('Setting nuget.config auth');
+        } else {
+            tl.debug('Not setting temp nuget.config auth');
```

Comment:
nit: add a message saying the new provider will handle auth, or something similar. So we know why we're not adding it.